unittest/client/src/ilm_control_unittests.cpp

```diff
@@ -0,0 +1,5071 @@
+    t_ilm_bool l_visibility = 1;
+    ASSERT_EQ(ILM_SUCCESS, ilm_layerGetVisibility(l_layerId, &l_visibility));
+
+    ASSERT_EQ(1, wl_proxy_marshal_flags_fake.call_count);
+    ASSERT_EQ(1, wl_display_roundtrip_queue_fake.call_count);
+}
+
+/** ================================================================================================
+ * @test_id             ilm_layerSetOpacity_wrongCtx
+ * @brief               Test case of ilm_layerSetOpacity() where ilm context wl.controller is null object
+ * @test_procedure Steps:
+ *                      -# Set the ilm context wl.controller is null object
+ *                      -# Calling the ilm_layerSetOpacity()
+ *                      -# Verification point:
+ *                         +# ilm_layerSetOpacity() must return ILM_FAILED
+ *                         +# wl_proxy_marshal_flags() not be called
+ *                         +# wl_display_flush() not be called
+ */
+TEST_F(IlmControlTest, ilm_layerSetOpacity_wrongCtx)
+{
+    ilm_context.wl.controller = nullptr;
+
+    t_ilm_layer l_layerId = 100;
+    t_ilm_float l_opacity = 1;
+    ASSERT_EQ(ILM_FAILED, ilm_layerSetOpacity(l_layerId, l_opacity));
+
+    ASSERT_EQ(0, wl_proxy_marshal_flags_fake.call_count);
+    ASSERT_EQ(0, wl_display_flush_fake.call_count);
+}
+
+/** ================================================================================================
+ * @test_id             ilm_layerSetOpacity_success
+ * @brief               Test case of ilm_layerSetOpacity() where ilm context is initilized and valid layer id
+ * @test_procedure Steps:
+ *                      -# Calling the ilm_layerSetOpacity()
+ *                      -# Verification point:
+ *                         +# ilm_layerSetOpacity() must return ILM_SUCCESS
+ *                         +# wl_proxy_marshal_flags() must be called once time
+ *                         +# wl_display_flush() must be called once time
+ */
+TEST_F(IlmControlTest, ilm_layerSetOpacity_success)
+{
+    t_ilm_layer l_layerId = 100;
+    t_ilm_float l_opacity = 1;
+    ASSERT_EQ(ILM_SUCCESS, ilm_layerSetOpacity(l_layerId, l_opacity));
+
+    ASSERT_EQ(1, wl_proxy_marshal_flags_fake.call_count);
+    ASSERT_EQ(1, wl_display_flush_fake.call_count);
+}
+
+/** ================================================================================================
+ * @test_id             ilm_layerGetOpacity_wrongVisibility
+ * @brief               Test case of ilm_layerGetOpacity() where input pOpacity is null pointer
+ * @test_procedure Steps:
+ *                      -# Set pOpacity is null pointer
+ *                      -# Calling the ilm_layerGetOpacity()
+ *                      -# Verification point:
+ *                         +# ilm_layerGetOpacity() must return ILM_FAILED
+ *                         +# wl_proxy_marshal_flags() not be called
+ *                         +# wl_display_roundtrip_queue() not be called
+ */
+TEST_F(IlmControlTest, ilm_layerGetOpacity_wrongVisibility)
```

Comment:
Should we change the testcase name here?
 ilm_layerGetOpacity_wrongVisibility ->  ilm_layerGetOpacity_InvalidArg or some other suitable name based on the test.

Reply:
done